spot_driver/spot_driver/spot_ros2.py

```diff
@@ -1466,245 +1470,216 @@ def handle_max_vel(self, request: SetVelocity.Request, response: SetVelocity.Res
+            if fb.safe_power_off_feedback.status.value == fb.safe_power_off_feedback.status.STATUS_POWERED_OFF:
                 return GoalResponse.SUCCESS
-            elif (
-                full_body_feedback.feedback.feedback_choice
-                == full_body_feedback.feedback.FEEDBACK_SELFRIGHT_FEEDBACK_SET
+            return GoalResponse.IN_PROGRESS
+
+        elif choice == fb.FEEDBACK_BATTERY_CHANGE_POSE_FEEDBACK_SET:
+            if fb.battery_change_pose_feedback.status.value == fb.battery_change_pose_feedback.status.STATUS_COMPLETED:
+                return GoalResponse.SUCCESS
+            if fb.battery_change_pose_feedback.status.value == fb.battery_change_pose_feedback.status.STATUS_FAILED:
+                return GoalResponse.FAILED
+            return GoalResponse.IN_PROGRESS
+
+        elif choice == fb.FEEDBACK_PAYLOAD_ESTIMATION_FEEDBACK_SET:
+            if fb.payload_estimation_feedback.status.value == fb.payload_estimation_feedback.status.STATUS_COMPLETED:
+                return GoalResponse.SUCCESS
+            if fb.payload_estimation_feedback.status.value == fb.payload_estimation_feedback.status.STATUS_SMALL_MASS:
+                return GoalResponse.SUCCESS
+            if fb.payload_estimation_feedback.status.value == fb.payload_estimation_feedback.status.STATUS_ERROR:
+                return GoalResponse.FAILED
+            return GoalResponse.IN_PROGRESS
+        elif choice == fb.FEEDBACK_CONSTRAINED_MANIPULATION_FEEDBACK_SET:
+            if (
+                fb.constrained_manipulation_feedback.status.value
+                == fb.constrained_manipulation_feedback.status.STATUS_RUNNING
             ):
-                if (
-                    full_body_feedback.feedback.selfright_feedback.status.value
-                    == full_body_feedback.feedback.selfright_feedback.status.STATUS_COMPLETED
-                ):
-                    return GoalResponse.SUCCESS
-                else:
-                    return GoalResponse.IN_PROGRESS
-            elif (
-                full_body_feedback.feedback.feedback_choice
-                == full_body_feedback.feedback.FEEDBACK_SAFE_POWER_OFF_FEEDBACK_SET
+                return GoalResponse.IN_PROGRESS
+            return GoalResponse.FAILED
+        return GoalResponse.IN_PROGRESS
+
+    def _process_synchronized_arm_command_feedback(self, feedback: ArmCommandFeedback) -> Optional[GoalResponse]:
+        if (
+            feedback.status.value == feedback.status.STATUS_COMMAND_OVERRIDDEN
+            or feedback.status.value == feedback.status.STATUS_COMMAND_TIMED_OUT
+            or feedback.status.value == feedback.status.STATUS_ROBOT_FROZEN
+            or feedback.status.value == feedback.status.STATUS_INCOMPATIBLE_HARDWARE
+        ):
+            return GoalResponse.FAILED
+
+        fb = feedback.feedback
+        choice = fb.feedback_choice
+
+        if choice == fb.FEEDBACK_ARM_CARTESIAN_FEEDBACK_SET:
```

Comment:
Where is the success return for this on trajectory complete?

Reply:
I realized later that this happens in robot command goal complete - that's a bit confusing when reading it.  Could this either be called something different to indicate it only returns on non-success or maybe just have robot command goal complete return whatever is returned here?